abe/resource_models/event_resources.py

```diff
@@ -133,6 +134,12 @@ def post(self):
         new_event = db.Event(**received_data)
         if new_event.labels == []:  # if no labels were given
             new_event.labels = ['unlabeled']
+        else:
+            for label in new_event.labels:
+                label_info = label_resources.LabelApi().get(id=label)
+                if type(label_info) is dict and 'protected' in label_info:
```

Comment:
Is `label_info` ever not a `dict`?

If there's no known reason for it not to be, and the unit tests pass without that check, I would be inclined towards simpler code (that omits the check).

Reply:
Yes, LabelApi get method is cringy...can return a dictionary, string, or list of dictionaries